crd-generator/api/src/test/java/io/fabric8/crd/generator/v1/JsonSchemaTest.java

```diff
@@ -178,15 +179,58 @@ void shouldExtractPropertiesSchemaFromExtractValueAnnotation() {
     assertNull(barProps.get("baz"));
   }
 
+  @Test
+  void shouldExtractPropertiesSchemaFromSchemaSwapAnnotations() {
```

Comment:
what's the difference in between this test and the one extracted in `SchemaSwapSpec` ? Or is it a leftover?

Reply:
There are two very similar tests: this one in crd-generator/api project, and another in crd-generator/test. 
They both perform the same set of modifications on the same schema, but the difference is that one runs in reflection context and the other on schema generated by Apt. 
In an ideal world, that shouldn't matter, but the sundr library is not yet perfect with annotation arguments that themselves are annotations, and these are sometimes presented as AnnotationRef and sometimes directly as the target annotation (SchemaSwap). Therefore, the code in AbstractJsonSchema needs to compensate for this by using two code paths, which both need to be tested.